gax-httpjson/src/test/java/com/google/api/gax/httpjson/HttpJsonClientInterceptorTest.java

```diff
@@ -224,8 +237,21 @@ public void testCustomInterceptor() throws ExecutionException, InterruptedExcept
     assertThat(headerValue).isEqualTo("headerValue");
 
     // Test that the custom interceptor was called
-    assertThat(interceptor.capturedStatusCode).isEqualTo(200);
+    // These two tests cases are guaranteed to respond back by the time `get()` is unblocked
     assertThat(interceptor.capturedResponseHeaders).isNotNull();
     assertThat(interceptor.capturedMessage).isEqualTo(request);
+
+    // TODO: We have no guarantee that `close()` will be called before `get()` is unblocked
+    // We must busy wait until the interceptor's `onClose()` is called
+    int attempts = 0;
+    while (interceptor.capturedStatusCode == 0) {
+      Thread.sleep(BUSY_WAIT_TIME);
+      attempts++;
+      if (attempts == NUM_RETRIES) {
+        break;
+      }
+    }
+
+    assertThat(interceptor.capturedStatusCode).isEqualTo(200);
```

Comment:
@blakeli0 Can you take a look at this: https://github.com/googleapis/gax-java/pull/1851#issuecomment-1299088772

I decided to just busy-waiting until the interceptor is done. Testing this locally and on GH actions it seems that the interceptor just barely misses the threshold needed for the `get()` call to block.

Reply:
Guava's `AbstractFuture.set()` docs: https://github.com/google/guava/blob/6d7e326b2cbfba5f19fc67859c0b3d4c45fab63f/guava/src/com/google/common/util/concurrent/AbstractFuture.java#L777

Reply:
On second thought, perhaps the future shouldn't unblock until `close()` is called? If we could store the message received from `onMessage` we can just set `future.set({message})` once `onClose` is called and get rid of the race condition.